File: app/src/main/java/com/namae0Two/khmeralternativekeyboard/MainActivity.java
package com.namae0Two.khmeralternativekeyboard;

import android.support.constraint.ConstraintLayout;
import android.support.constraint.ConstraintSet;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.namae0Two.khmeralternativekeyboard.view.KeyboardView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        ConstraintLayout mainlayout = (ConstraintLayout)findViewById(R.id.mainlayout);

        //
        KeyboardView keyboardView = new KeyboardView(getApplicationContext());


        ConstraintSet s = new ConstraintSet();
        s.clone(mainlayout);

        s.connect(keyboardView.getId(), ConstraintSet.TOP, mainlayout.getId(), ConstraintSet.TOP, 0);
        s.connect(keyboardView.getId(), ConstraintSet.END, mainlayout.getId(), ConstraintSet.END, 0);
        s.connect(keyboardView.getId(), ConstraintSet.BOTTOM, mainlayout.getId(), ConstraintSet.BOTTOM, 0);
        s.connect(keyboardView.getId(), ConstraintSet.START, mainlayout.getId(), ConstraintSet.START, 0);
        mainlayout.addView(keyboardView);

        s.applyTo(mainlayout);


    }
}

File: app/src/main/java/com/namae0Two/khmeralternativekeyboard/MainActivity.kt
package com.namae0Two.khmeralternativekeyboard

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.namae0Two.khmeralternativekeyboard.config.ViewConfig
import com.namae0Two.khmeralternativekeyboard.database.DictionaryWordDatabase
import com.namae0Two.khmeralternativekeyboard.database.ioThread

class MainActivity : AppCompatActivity() {
    private var database: DictionaryWordDatabase? = null

    private var viewConfig: ViewConfig? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewConfig = ViewConfig.getInstance(applicationContext)

        if (database == null) {
            database = DictionaryWordDatabase.getInstance(application)
        }
        ioThread {
            //invoke onCreate
            database!!.dictionaryWordDao().getAll()
        }
    }

//    override fun onStop() {
////        Log.d("Filling", "Deleting Database")
////        applicationContext.deleteDatabase("dictionary.db")
//
//        super.onStop()
//    }
}
